network/crl.go

```diff
@@ -331,23 +334,36 @@ func (v DefaultCRLVerifier) getCachedOrFetch(url string, allowLocal bool, issuer
 		return nil, err
 	}
 
+	revokedCertificates := make(map[*big.Int]pkix.RevokedCertificate, len(crl.TBSCertList.RevokedCertificates))
+	for _, cert := range crl.TBSCertList.RevokedCertificates {
+		revokedCertificates[cert.SerialNumber] = cert
+	}
+
 	if crl.TBSCertList.NextUpdate.Before(time.Now()) {
 		log.Warnf("CRL: CRL at %s is outdated", url)
 		return nil, ErrOutdatedCRL
 	}
 
+	// We don't need this array anymore as all revoked certificates are now inside the `map`
+	crl.TBSCertList.RevokedCertificates = make([]pkix.RevokedCertificate, 0)
+
+	cacheItem := &CRLCacheItem{
+		Fetched:             time.Now(),
+		CRL:                 *crl,
```

Comment:
do we need to do copy of whole list instead of copying pointer?

Reply:
Changed struct to contain pointer instead, and put `CRL: crl` here